src/lib/include/x86_vector_triple.h

```diff
@@ -0,0 +1,51 @@
+#ifndef DISTOPIA_X86_VECTOR_TRIPLE
+#define DISTOPIA_X86_VECTOR_TRIPLE
+
+#include "distopia_type_traits.h"
+#include "x86_swizzle.h"
+#include "x86_vectors.h"
+#include "x86_tgintrin.h"
+
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT> = 0,
+          EnableIfFloating<ScalarT> = 0, EnableIfMatching<VectorT, ScalarT> = 0>
+class VectorTriple {
+public:
+  VectorT a;
+  VectorT b;
+  VectorT c;
+  // should this be size_t or uintptr_t?
+  constexpr static intptr_t nvals_per_pack = ValuesPerPack<VectorT>::value_p;
+
+  // from 3 SIMD Vector datatypes eg __m128 or __m128d
+  inline VectorTriple(VectorT a, VectorT b, VectorT c) : a(a), b(b), c(c) {}
+
+  // load from a vector of ScalarT eg float* or double * as constructor
+  inline VectorTriple(ScalarT *source) {
+    a = load_p(source);
```

Comment:
these macros are not working from `x86_tgintrin.h` Im probably doing something wrong.

Reply:
`load_p` is a little bit strange because it doesn’t have enough information (from the type of the argument) to determine the result type. For example `load_p(float const*)` could return either `__m128` or `__m256`. You have to specialize it explicitly: `load_p<VectorT>(source)` should work.

Reply:
still no luck.. Ive pushed so you can see what I have done.

Reply:
Still no luck ... I have pushed so you can see what I have done.